src/webgpu/shader/execution/expression/call/builtin/subgroupBroadcast.spec.ts

```diff
@@ -0,0 +1,316 @@
+export const description = `
+Execution tests for subgroupBroadcast
+
+Note: There is a lack of portability for non-uniform execution so these tests
+restrict themselves to uniform control flow.
+Note: There is no guaranteed mapping between subgroup_invocation_id and
+local_invocation_index. Tests should avoid assuming there is.
+`;
+
+import { makeTestGroup } from '../../../../../../common/framework/test_group.js';
+import { keysOf, objectsToRecord } from '../../../../../../common/util/data_tables.js';
+import { iterRange } from '../../../../../../common/util/util.js';
+import { GPUTest } from '../../../../../gpu_test.js';
+import {
+  kConcreteNumericScalarsAndVectors,
+  Type,
+  VectorType,
+  elementTypeOf,
+} from '../../../../../util/conversion.js';
+
+export const g = makeTestGroup(GPUTest);
+
+const kDataTypes = objectsToRecord(kConcreteNumericScalarsAndVectors);
+
+const kWGSizes = [
+  [4, 1, 1],
+  [8, 1, 1],
+  [16, 1, 1],
+  [32, 1, 1],
+  [64, 1, 1],
+  [128, 1, 1],
+  [256, 1, 1],
+  [1, 4, 1],
+  [1, 8, 1],
+  [1, 16, 1],
+  [1, 32, 1],
+  [1, 64, 1],
+  [1, 128, 1],
+  [1, 256, 1],
+  [1, 1, 4],
+  [1, 1, 8],
+  [1, 1, 16],
+  [1, 1, 32],
+  [1, 1, 64],
+  [3, 3, 3],
+  [4, 4, 4],
+  [16, 16, 1],
+  [16, 1, 16],
+  [1, 16, 16],
+  [15, 3, 3],
+  [3, 15, 3],
+  [3, 3, 15],
+] as const;
+
+g.test('data_types')
+  .desc('Tests broadcast of data types')
```

Comment:
the values used are value-conversions of the subgroup invocation id.

Add a TODO to check large magnitude values for the scalar type under test.  e.g. 1e25 for f32.